### ISSUE TITLE
Cross Origin Resource Sharing support

### ISSUE THREAD

**Issue by jhampson-dbre** [python-restx/flask-restx#183]
How can I enable CORS headers for Flask-restx?

I have a need to enable CORS headers on my API response. Digging through the flask-restx code base, I noticed there is an undocumented [cors submodule](https://github.com/python-restx/flask-restx/blob/014eb9591e61cd3adbbd29a38b76df6a688f067b/tests/test_cors.py).

I see in #64 , there is a recommendation to use Flask-CORS for adding CORS headers, so I'm wondering if the undocumented cors functionality in restx should be avoided?

I have lightly tested it with the following:

```python
from flask_restx import Namespace, Resource, reqparse, fields, cors

ns = Namespace('inventories', description="Inventory",
               decorators=[cors.crossdomain(origin="*")])
```

And it seems to return the headers as expected

```
 access-control-allow-methods: GET, HEAD 
 access-control-allow-origin: * 
 access-control-max-age: 21600 
```

Based on the tests in test_cors.py, I originally tried
 
```python
@ns.route("/oem/")
class OEMInventory(Resource):
    @cors.crossdomain(origin="*")
    @ns.marshal_with(oem_inventory_model)
    def get(self):
      ...
      return [data for host, data in oem_hosts.items()]
```
But that seem to break marshal_with, as I started getting errors about "list is not a valid response object".

**Comment by s-weigand** [comment]
@jhampson-dbre Thanks a lot for showing me how to enable `CORS`.  🎉

**Comment by jhampson-dbre** [comment]
@s-weigand Glad this helped! Since I never got any response on if the undocumented built-in cors should be avoided, I ended up using Flask-CORS. Even with this, I'm still having trouble getting the decorator to work on individual routes (same with other Flask extensions I've tried to use decorators with). Fortunately, I only had one route in this namespace so I could use the `decorators` property of `Namespace` to apply it. 

```python
from flask_restx import Namespace, Resource, reqparse, fields, inputs
from flask_cors import cross_origin

ns = Namespace('inventories', description="OEM Inventory",
               decorators=[cross_origin()])
```

If anyone gets the decorator working, I'd be interested to see that implementation.
